internal/cmd/egctl/translate.go

```diff
@@ -276,7 +276,7 @@ func translateGatewayAPIToIR(resources *gatewayapi.Resources) (*gatewayapi.Trans
 	}
 
 	t := &gatewayapi.Translator{
-		GatewayControllerName:   egv1a1.GatewayControllerName,
+		GatewayControllerName:   string(resources.GatewayClass.Spec.ControllerName),
```

Comment:
probably good to check if `resources.GatewayClass` is `nil` or not, before dereferncing it

Reply:
https://github.com/envoyproxy/gateway/blob/67008f83dc7e10df2b5c94c578a9379c3b688be7/internal/cmd/egctl/translate.go#L691-L694

did it

Reply:
won't this crash if no `GatewayClass` is provided ?

Reply:
@arkodg yes, gateway class checked before translate  resouce 
https://github.com/envoyproxy/gateway/blob/67008f83dc7e10df2b5c94c578a9379c3b688be7/internal/cmd/egctl/translate.go#L332-L334